db/seeds.rb

```diff
@@ -97,503 +65,390 @@ def terms_of_service_text
-        em.associated_equipment_models = EquipmentModel.all.sample(6)
-      end
-    end
-    puts "\n#{entered_num} equipment model records successfully created!"
-  else
-    puts "\nPlease enter a whole number greater than 0."
-    entered_num = STDIN.gets.chomp.to_i
+def generate_req
+  Requirement.create! do |req|
+    req.equipment_models = EquipmentModel.all.sample(R.rand(1..3))
+    req.contact_name = Faker::Name.name
+    req.contact_info = Faker::PhoneNumber.short_phone_number
+    req.notes = Faker::HipsterIpsum.paragraph(4)
+    req.description = Faker::HipsterIpsum.sentence
   end
-else
-  puts "\n***To seed equipment objects, blackout dates,
-   requirements, reservations, etc, you must create
-   at least one equipment model. Exiting seed script...***\n\n"
-  exit
 end
 
-# Requirement generation
-# ============================================================================
-
-if ENV["minimal"]
-  entered_num = 0
-else
-  entered_num = ask_for_records("Requirement")
+def generate_checkin
+  CheckinProcedure.create! do |chi|
+    chi.step = Faker::HipsterIpsum.sentence
+    chi.equipment_model_id = EquipmentModel.all.sample.id
+  end
 end
 
-unless entered_num == 0
-  if entered_num.integer? && entered_num > 0
-    progress = ProgressBar.create(format: progress_str, total: entered_num)
-    requirement = entered_num.times.map do
-      progress.increment
-      Requirement.create! do |req|
-        req.equipment_models = equipment_model.sample(r.rand(1..3))
-        req.contact_name = Faker::Name.name
-        req.contact_info = Faker::PhoneNumber.short_phone_number
-        req.notes = Faker::HipsterIpsum.paragraph(4)
-        req.description = Faker::HipsterIpsum.sentence
-      end
-    end
-    puts "\n#{entered_num} requirement records successfully created!"
-  else
-    puts "\nPlease enter a whole number greater than 0."
-    entered_num = STDIN.gets.chomp.to_i
+def generate_checkout
+  CheckoutProcedure.create! do |chi|
+    chi.step = Faker::HipsterIpsum.sentence
+    chi.equipment_model_id = EquipmentModel.all.sample.id
   end
 end
 
-# CheckinProcedure generation
-# ============================================================================
+def generate_blackout
+  Blackout.create! do |blk|
+    blk.start_date = time_rand(Time.now + 1.year)
+    blk.end_date = time_rand(blk.start_date.to_time, blk.start_date.next_week.to_time)
+    blk.notice = Faker::HipsterIpsum.paragraph(2)
+    blk.created_by = User.first.id
+    blk.blackout_type = ['soft', 'hard'].sample
+  end
+end
 
-if ENV["minimal"]
-  entered_num = 3
-else
-  entered_num = ask_for_records("CheckinProcedure")
+def generate_reservation
+  random_time = time_rand(Time.now, Time.now + 2.months)
+  res = Reservation.new
+  res.reserver_id = User.all.sample.id
+  res.checkout_handler_id = User.where("role = ? OR role = ? OR role = ?","checkout", "admin", "superuser").all.sample.id
```

Comment:
Wouldn't this be setting checkin and checkout handlers regardless of whether or not the reservation is checked in / checked out? We should probably move this after that determination and only set these if necessary.